lib/posix/signal.c

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2023 Meta
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include "signal.h"
+
+#include <errno.h>
+#include <string.h>
+
+#include <zephyr/posix/signal.h>
+
+#define SIGNO_WORD_IDX(_signo) ((signo - 1) / _NSIG_BPW)
+#define SIGNO_WORD_BIT(_signo) ((signo - 1) % _NSIG_BPW)
+
+int sigemptyset(sigset_t *set)
+{
+	memset(set, 0, sizeof(*set));
+
+	return 0;
+}
+
+int sigfillset(sigset_t *set)
+{
+	memset(set, -1, sizeof(*set));
```

Comment:
memset is byte oriented while the set array is word oriented. A loop here would likely be faster than memset (not that it's a huge gain)